sql/api/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -208,22 +206,33 @@ object DataType {
   }
 
   // NOTE: Map fields must be sorted in alphabetical order to keep consistent with the Python side.
-  private[sql] def parseDataType(json: JValue): DataType = json match {
+  private[sql] def parseDataType(
+      json: JValue,
+      fieldPath: String = "",
+      collationsMap: Map[String, String] = Map.empty): DataType = json match {
     case JString(name) =>
-      nameToType(name)
+      collationsMap.get(fieldPath) match {
```

Comment:
I believe names in the schema are case insensitive (or at least can be). Can this cause us a problem here? I guess not because we are building and reading this from the same instance of a schema, but want to double check